File: example-android-project/app/src/main/java/io/github/gofaith/faithdroid/UI/DemoObjectFragment.java
package io.github.gofaith.faithdroid.UI;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import faithdroid.Faithdroid;
import io.github.gofaith.faithdroid.FViews.FView;
import io.github.gofaith.faithdroid.FViews.FViewPager;


public  class DemoObjectFragment extends Fragment {
    public static final String ARG_OBJECT = "object";
    private FViewPager fviewPager;

    public DemoObjectFragment setFviewPager(FViewPager fviewPager) {
        this.fviewPager=fviewPager;
        return this;
    }
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        Bundle args = getArguments();
        int i=args.getInt(ARG_OBJECT);
        if (fviewPager == null) {
            return null;
        }
        if (fviewPager.onCreateView != null && !fviewPager.onCreateView.equals("")) {
            Log.d("TAG", "onCreateView: 1");
            String vid = Faithdroid.triggerEventHandler(fviewPager.onCreateView, String.valueOf(i));
            if (fviewPager.parentController.viewmap.containsKey(vid)) {
                Log.d("TAG", "onCreateView: 2");
                FView cview = fviewPager.parentController.viewmap.get(vid);
                if (cview != null) {
                    Log.d("TAG", "onCreateView: 3");
                    return cview.view;
                }
            }
        }
        if (fviewPager.pages.size() <= i) {
            return null;
        }
        FViewPager.FPage p = fviewPager.pages.get(i);
        View rootView=fviewPager.parentController.viewmap.get(Faithdroid.triggerEventHandler(p.vid,"")).view;
        return rootView;
    }
}

File: android/app/src/main/java/io/github/stevenzack/faithdroidapp/UI/DemoObjectFragment.kt
package io.github.stevenzack.faithdroidapp.UI

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import faithdroid.Faithdroid
import io.github.stevenzack.faithdroidapp.FViews.FViewPager


class DemoObjectFragment : Fragment() {
    private var fviewPager: FViewPager? = null

    fun setFviewPager(fviewPager: FViewPager): DemoObjectFragment {
        this.fviewPager = fviewPager
        return this
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val args = arguments
        val i = args!!.getInt(ARG_OBJECT)
        if (fviewPager == null) {
            return null
        }
        if (fviewPager!!.onCreateView != null && !fviewPager!!.onCreateView.equals("")) {
            Log.d("TAG", "onCreateView: 1")
            val vid = Faithdroid.triggerEventHandler(fviewPager!!.onCreateView, i.toString())
            if (fviewPager!!.parentController!!.viewmap.containsKey(vid)) {
                Log.d("TAG", "onCreateView: 2")
                val cview = fviewPager!!.parentController!!.viewmap.get(vid)
                if (cview != null) {
                    Log.d("TAG", "onCreateView: 3")
                    return cview!!.view
                }
            }
        }
        if (fviewPager!!.pages.size <= i) {
            return null
        }
        val p = fviewPager!!.pages.get(i)
        return fviewPager!!.parentController!!.viewmap.get(Faithdroid.triggerEventHandler(p.vid, ""))!!.view
    }

    companion object {
        val ARG_OBJECT = "object"
    }
}
